File: Java_Programming/SingleLevel.java
class Base
{
    public int i, j;
    public Base(){
        System.out.println("Inside Base Constructor");
    }
    public void fun(){
        System.out.println("Inside Base fun");
    }
    public void gun(){
        System.out.println("Inside Base gun");
    }
}

class Derived extends Base{
    public int x, y;
    public Derived()
    {
        System.out.println("Inside Derived Constructor");
    }
    public void sun()
    {
        System.out.println("Inside Derived sun");
    }
}

class SingleLevel
{
    public static void main(String A[]) {
        Derived dobj = new Derived();
        dobj.fun();
        dobj.gun();
        dobj.sun();
    }
}

File: Java_Programming/SingleLevel.cpp
#include<iostream>
using namespace std;
class Base
{
    public:
    int i,j;
    Base()
    {
        cout<<"Inside Base Constructor.";
    }
    ~Base()
    {
        cout<<"Inside Base Destructor.\n";
    }
    void fun()
    {
        cout<<"Inside Base fun\n";
    }
    void gun()
    {
        cout<<"Inside Base gun\n";
    }
};

class Derived : public Base
{
    public:
        int x,y;
        Derived()
        {
            cout<<"Inside Derived Constructor.\n";
        }
        ~Derived()
        {
            cout<<"Inside Derived Destructor.\n";
        }
        void sun()
        {
            cout<<"Inside Derived sun\n";
        }
};

int main()
{
    cout<<sizeof(Base)<<"\n";
    cout<<sizeof(Derived)<<"\n";

    return 0;
}